src/http/cookie.cr

```diff
@@ -8,30 +8,35 @@ module HTTP
     property value : String
     property path : String
     property expires : Time?
+    property max_age : Time::Span?
     property domain : String?
     property secure : Bool
     property http_only : Bool
     property extension : String?
+    property creation_time : Time
```

Comment:
I don't think it makes sense to have `creation_time` writable, and I'm not sure if it should even be readable externally.

To have a customizeable time reference, I would rather add an optional argument to `expiration_time` and `expired?`. It could be called `current_time`  or `time_reference`. That would be a better API, because you don't need to change the Cookie's internal data to check its expiration at a certain point of time.

Reply:
I think I set this up as you described, though I'm happy to rewrite. I added tests for using reference_time in specs as well, though the spec naming may be too verbose.

Reply:
Looks good!

However on a second thought, it would probably make sense to have `creation_time` readable just to be transparent about it. And I'd also suggest to add `creation_time` as an optional named parameter to `initialize`. This allows for time-independent testing as well as (re-)creating a Cookie instance which was retrieved at a different point in time, maybe for serialization.
I think it makes generally sense to allow all references to `Time.new` to be replaced with an optional parameter. So maybe even `expired?` should get a second, optional named parameter...